### ISSUE TITLE
Raise error when result.output is none

### ISSUE THREAD

**Issue by nicolocin** [nipype/pydra#290]
`output` can be `None` when output is not set in `set_output` or if one of the tasks doesn't return anything.

The error currently looks like something like which isn't very descriptive.

```
<ipython-input-1-a0fe8ea3a54b> in run_workflow(wf, plugin, plugin_args, specfile)
    339     cwd = os.getcwd()
    340     with pydra.Submitter(plugin=plugin, **plugin_args) as sub:
--> 341         sub(runnable=wf)
    342     results = wf.result(return_inputs=True)
    343     os.chdir(cwd)

/usr/local/anaconda3/lib/python3.7/site-packages/pydra/engine/submitter.py in __call__(self, runnable, cache_locations, rerun)
     55             ]
     56         if is_workflow(runnable) and runnable.state is None:
---> 57             self.loop.run_until_complete(self.submit_workflow(runnable, rerun=rerun))
     58         else:
     59             self.loop.run_until_complete(self.submit(runnable, wait=True, rerun=rerun))

/usr/local/anaconda3/lib/python3.7/asyncio/base_events.py in run_until_complete(self, future)
    582             raise RuntimeError('Event loop stopped before Future completed.')
    583 
--> 584         return future.result()
    585 
    586     def stop(self):

/usr/local/anaconda3/lib/python3.7/site-packages/pydra/engine/submitter.py in submit_workflow(self, workflow, rerun)
     70                 await self.worker.run_el(workflow, rerun=rerun)
     71             else:
---> 72                 await workflow._run(self, rerun=rerun)
     73         else:  # could be a tuple with paths to pickle files wiith tasks and inputs
     74             ind, wf_main_pkl, wf_orig = workflow

/usr/local/anaconda3/lib/python3.7/site-packages/pydra/engine/core.py in _run(self, submitter, rerun, **kwargs)
    924                 self.hooks.post_run_task(self, result)
    925                 self.audit.finalize_audit(result=result)
--> 926                 save(odir, result=result, task=self)
    927                 os.chdir(cwd)
    928         self.hooks.post_run(self, result)

/usr/local/anaconda3/lib/python3.7/site-packages/pydra/engine/helpers.py in save(task_path, result, task, name_prefix)
    129             if task_path.name.startswith("Workflow"):
    130                 # copy files to the workflow directory
--> 131                 result = copyfile_workflow(wf_path=task_path, result=result)
    132             with (task_path / f"{name_prefix}_result.pklz").open("wb") as fp:
    133                 cp.dump(result, fp)

/usr/local/anaconda3/lib/python3.7/site-packages/pydra/engine/helpers.py in copyfile_workflow(wf_path, result)
    139 def copyfile_workflow(wf_path, result):
    140     """ if file in the wf results, the file will be copied to the workflow directory"""
--> 141     for field in attr_fields(result.output):
    142         value = getattr(result.output, field.name)
    143         new_value = _copyfile_single_value(wf_path=wf_path, value=value)

/usr/local/anaconda3/lib/python3.7/site-packages/pydra/engine/specs.py in attr_fields(x)
      6 
      7 def attr_fields(x):
----> 8     return x.__attrs_attrs__
      9 
     10 

AttributeError: 'NoneType' object has no attribute '__attrs_attrs__'
```

**Comment by djarecka** [comment]
we actually should also raise error in some common cases, before it gets here.  for example, for a workflow, we can check if `set_output` is used, and if not raise an error, before it gets here

**Comment by djarecka** [comment]
closed with #303
